sympy/printing/tests/test_pycode.py

```diff
@@ -92,6 +105,26 @@ def test_SciPyPrinter():
     smat = SparseMatrix(2, 5, {(0, 1): 3})
     assert p.doprint(smat) == 'scipy.sparse.coo_matrix([3], ([0], [1]), shape=(2, 5))'
     assert 'scipy.sparse' in p.module_imports
+    assert p.doprint(erf(x)) == 'scipy.special.erf(x)'
+    assert p.doprint(erfc(x)) == 'scipy.special.erfc(x)'
+    assert p.doprint(besselj(x, y)) == 'scipy.special.jv(x, y)'
+    assert p.doprint(bessely(x, y)) == 'scipy.special.yv(x, y)'
+    assert p.doprint(besseli(x, y)) == 'scipy.special.iv(x, y)'
+    assert p.doprint(besselk(x, y)) == 'scipy.special.kv(x, y)'
+    assert p.doprint(factorial(x)) == 'scipy.special.factorial(x, exact=true)'
+    assert p.doprint(gamma(x)) == 'scipy.special.gamma(x)'
+    assert p.doprint(loggamma(x)) == 'scipy.special.gammaln(x)'
+    assert p.doprint(digamma(x)) == 'scipy.special.psi(x)'
```

Comment:
I don't think that the any symbolic object for `digamma` exists for SymPy, but a redirective to `polygamma`
It would be more tricky to fix, but it would either be adding `_print_polygamma` to the scipy printer and dispatching the trivial case with `psi`, like how we treat `sqrt` vs `pow`
Or making a change to the sympy `digamma` function itself, to be a full symbolic node which automatically evaluates to `polygamma` if `evaluate=True`, but remain `digamma` if `evaluate=False`.

Reply:
Yes I see there's a problem here, thank you for pointing this out.

Reply:
@sylee957 should I make changes in the digamma function to evaluate digamma if evaluate=False? Adding _print_polygamma seems a little complicated to me. If I take reference from sqrt and pow, many functions need to be added for digamma as well?

Reply:
I would need help with  _print_polygamma or making changes to the existing function.

Reply:
I have the diff like below, which can resolve the issue related to digamma without adding `_print_polygamma`, but one limitation is that things will not be printed in `digamma` unless the automatic evaluation is manually disabled.

```
$ git diff
diff --git a/sympy/core/tests/test_args.py b/sympy/core/tests/test_args.py
index 38df142210..0d7b2ad263 100644
--- a/sympy/core/tests/test_args.py
+++ b/sympy/core/tests/test_args.py
@@ -2278,6 +2278,11 @@ def test_sympy__functions__special__gamma_functions__polygamma():
     assert _test_args(polygamma(x, 2))


+def test_sympy__functions__special__gamma_functions__digamma():
+    from sympy.functions.special.gamma_functions import digamma
+    assert _test_args(digamma(x))
+
+
 def test_sympy__functions__special__gamma_functions__uppergamma():
     from sympy.functions.special.gamma_functions import uppergamma
     assert _test_args(uppergamma(x, 2))
diff --git a/sympy/functions/special/gamma_functions.py b/sympy/functions/special/gamma_functions.py
index d9bb918e32..d596ad0a90 100644
--- a/sympy/functions/special/gamma_functions.py
+++ b/sympy/functions/special/gamma_functions.py
@@ -965,7 +965,7 @@ def _sage_(self):
         return sage.log_gamma(self.args[0]._sage_())


-def digamma(x):
+class digamma(Function):
     r"""
     The digamma function is the first derivative of the loggamma function i.e,

@@ -993,7 +993,10 @@ def digamma(x):
     .. [2] http://mathworld.wolfram.com/DigammaFunction.html
     .. [3] http://functions.wolfram.com/GammaBetaErf/PolyGamma2/
     """
-    return polygamma(0, x)
+    def __new__(cls, x, evaluate=True):
+        if evaluate:
+            return polygamma(0, x, evaluate=evaluate)
+        return super(digamma, cls).__new__(cls, x, evaluate=False)


 def trigamma(x):
diff --git a/sympy/printing/pycode.py b/sympy/printing/pycode.py
index 2d908b4236..744132076a 100644
--- a/sympy/printing/pycode.py
+++ b/sympy/printing/pycode.py
@@ -818,6 +818,7 @@ def _print_CodegenArrayElementwiseAdd(self, expr):
     'factorial': 'factorial',
     'gamma': 'gamma',
     'loggamma': 'gammaln',
+    'polygamma': 'polygamma',
     'digamma': 'psi',
     'RisingFactorial': 'poch',
     'jacobi': 'eval_jacobi',
diff --git a/sympy/printing/tests/test_pycode.py b/sympy/printing/tests/test_pycode.py
index 30af61da9e..cb54fe751a 100644
--- a/sympy/printing/tests/test_pycode.py
+++ b/sympy/printing/tests/test_pycode.py
@@ -114,6 +114,9 @@ def test_SciPyPrinter():
     assert p.doprint(besselk(x, y)) == 'scipy.special.kv(x, y)'
     assert p.doprint(factorial(x)) == 'scipy.special.factorial(x)'
     assert p.doprint(gamma(x)) == 'scipy.special.gamma(x)'
+    assert p.doprint(polygamma(x, y)) == 'scipy.special.polygamma(x, y)'
+    assert p.doprint(digamma(x)) == 'scipy.special.polygamma(0, x)'
+    assert p.doprint(digamma(x, evaluate=False)) == 'scipy.special.psi(x)'
     assert p.doprint(loggamma(x)) == 'scipy.special.gammaln(x)'
     assert p.doprint(RisingFactorial(x, y)) == 'scipy.special.poch(x, y)'
     assert p.doprint(jacobi(1, x, y, z, evaluate=False)) == 'scipy.special.eval_jacobi(1, x, y, z)'

```